pageserver/src/layered_repository.rs

```diff
@@ -1938,41 +1938,71 @@ impl LayeredTimeline {
         Ok(new_path)
     }
 
+    ///
+    /// Collect a bunch of Level 0 layer files, and compact and reshuffle them as
+    /// as Level 1 files.
```

Comment:
Should we mention the collect criteria in this doc? E.g "[This is done by] collecting contiguous sequences of level 0 files, so that the end LSN of previous delta file is equal to the start of the next one."